js/ui/shellEntry.js

```diff
@@ -184,6 +179,28 @@ var OverviewEntry = GObject.registerClass({
         global.stage.connect('notify::key-focus', this._onStageKeyFocusChanged.bind(this));
     }
 
+    _updateSearchEntryText() {
+
+        let entryText;
+
+        if (this._parentalControlsManager.isContentTypeAllowed("x-scheme-handler/http")) {
```

Comment:
I think it would probably be better to check whether we have an internet search provider here given that is what decides whether we can search the internet.

Reply:
I think the shell can have the internet provider but it can be blocked via parental controls. I am also not sure what plumbing might be required here to actually decide if the entry text needs to be updated.

What is wrong with the current approach i.e. querying `x-scheme-handler/http` to decide?

Reply:
You are correct that we can have an internet search provider although it can be blocked, but I think the problem here is that we are making assumptions again on how browsers are blocked via parental controls (which is done by a different code/component).

It would be the same as only creating an Internet search provider if the content type is not blocked, which is not what we do, we create it and ask the MctAppFilter whether it is blocked via appinfo (which also checks executable path and flatpak app ID apart from content types to decide whether something is blacklisted).

To clarify, the code is not wrong, it will work with the current plans, but it is making assumptions that I dont think should be here.

Reply:
My suggestion is that we check if the internet search provider is blocked (not registered) and change the text accordingly.

Reply:
See also related discussions on #parental-controls@slack.